cmd/tnijto/main.go

```diff
@@ -1,3 +1,66 @@
 package main
 
-func main() {}
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	utils "github.com/turbaszek/tnijto/pkg"
+	"log"
+	"net/http"
+	"time"
+)
+
+var config = utils.NewConfig()
+
+func main() {
+	router := mux.NewRouter()
+
+	router.Handle("/", http.FileServer(http.Dir("./static")))
+	router.HandleFunc("/new", submitNewLinkHandler)
+
+	router.Use(utils.LoggingMiddleware)
+	router.NotFoundHandler = utils.Handle404()
+
+	srv := &http.Server{
+		Handler: router,
+		Addr:    ":1317",
+		// Good practice: enforce timeouts for servers you create!
+		WriteTimeout: 15 * time.Second,
+		ReadTimeout:  15 * time.Second,
+	}
+
+	log.Fatal(srv.ListenAndServe())
+}
+
+// Link represents the link
+type Link struct {
+	URL          string
+	Name         string
+	GeneratedURL string
+}
+
+func submitNewLinkHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Redirect(w, r, "/", http.StatusSeeOther)
+		return
+	}
+
+	link := r.FormValue("link")
+	name := r.FormValue("name")
+	generated := fmt.Sprintf("https://%s/%s", config.Hostname, name)
+
+	linkResponse := Link{link, name, generated}
+
+	js, err := json.Marshal(linkResponse)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
```

Comment:
I'd extract error handling to function or some dedicated component. WDYT?

Reply:
I thought about it, but not sure if necessary yet